integration-tests/smoke/ccip_test.go

```diff
@@ -168,105 +151,233 @@ func TestTokenTransfer(t *testing.T) {
-					ExtraArgs:    nil,
-				})
-				expectedSeqNum[dest] = msgSentEvent.SequenceNumber
-			} else {
-				msgSentEvent := ccdeploy.TestSendRequest(t, e, state, src, dest, false, router.ClientEVM2AnyMessage{
-					Receiver:     receiver,
-					Data:         data,
-					TokenAmounts: nil,
-					FeeToken:     feeToken,
-					ExtraArgs:    nil,
-				})
-				expectedSeqNum[dest] = msgSentEvent.SequenceNumber
+			msg := router.ClientEVM2AnyMessage{
+				Receiver:     common.LeftPadBytes(scenario.receiver.Bytes(), 32),
+				Data:         scenario.data,
+				TokenAmounts: scenario.tokenAmounts,
+				FeeToken:     common.HexToAddress("0x0"),
+				ExtraArgs:    nil,
 			}
-		}
+
+			// Send the message
+			msgSentEvent := ccdeploy.TestSendRequest(t, e, state, scenario.srcChain, scenario.dstChain, false, msg)
+			expectedSeqNum[scenario.dstChain] = msgSentEvent.SequenceNumber
+
+			// Wait for commit and execution
+			ccdeploy.ConfirmCommitForAllWithExpectedSeqNums(t, e, state, expectedSeqNum, startBlocks)
+			ccdeploy.ConfirmExecWithSeqNrForAll(t, e, state, expectedSeqNum, startBlocks)
+
+			// Fetch final balances and assert
+			for tokenAddress, totalAmount := range totalAmountsTransferred {
+				dstToken := getTokenByAddress(dstToken1, dstToken2, tokenAddress)
+				require.NotNil(t, dstToken, "Destination token not found for address %s", tokenAddress.Hex())
+
+				finalBalance, err := dstToken.BalanceOf(nil, scenario.receiver)
+				require.NoError(t, err)
+
+				initialBalance := initialBalances[dstToken.Address()]
+				expectedBalance := new(big.Int).Add(initialBalance, totalAmount)
+
+				require.Equal(t, expectedBalance, finalBalance, "Incorrect balance for token %s", dstToken.Address().Hex())
+			}
+		})
 	}
+}
 
-	// Wait for all commit reports to land.
-	ccdeploy.ConfirmCommitForAllWithExpectedSeqNums(t, e, state, expectedSeqNum, startBlocks)
+func getTokenByAddress(dstToken1, dstToken2 *burn_mint_erc677.BurnMintERC677, tokenAddress common.Address) *burn_mint_erc677.BurnMintERC677 {
+	if dstToken1.Address() == tokenAddress {
+		return dstToken1
+	} else if dstToken2.Address() == tokenAddress {
+		return dstToken2
+	}
+	return nil
+}
 
-	// After commit is reported on all chains, token prices should be updated in FeeQuoter.
-	for dest := range e.Chains {
-		linkAddress := state.Chains[dest].LinkToken.Address()
-		feeQuoter := state.Chains[dest].FeeQuoter
-		timestampedPrice, err := feeQuoter.GetTokenPrice(nil, linkAddress)
-		require.NoError(t, err)
-		require.Equal(t, ccdeploy.MockLinkPrice, timestampedPrice.Value)
+// Helper function to determine the destination token
+func getDestinationToken(scenario struct {
+	name         string
+	srcChain     uint64
+	dstChain     uint64
+	tokenAmounts []router.ClientEVMTokenAmount
+	receiver     common.Address
+	data         []byte
+}, tenv ccdeploy.DeployedEnv, srcToken1, dstToken1, srcToken2, dstToken2 *burn_mint_erc677.BurnMintERC677, tokenAddress common.Address) *burn_mint_erc677.BurnMintERC677 {
```

Comment:
Code smell to take an anonymous struct, better to just pass in the args you need (looks like srcChain, dstChain)

Reply:
Good idea, I missed to remove this during a refactor